lib/publiccloud/provider.pm

```diff
@@ -230,37 +220,87 @@ sub create_instance {
     $args{instance_type} //= get_var('PUBLIC_CLOUD_INSTANCE_TYPE');
     $args{image} //= $self->get_image_id();
 
+    my @vms = $self->terraform_apply(count => '1');
+    my $instance = $vms[0];
+    record_info('INFO', "VM created '" . Dumper $instance);
+
+    my $timeout = 60;
+    while ($timeout > 0) {
+        last if script_run("nc -vz -w 1 $instance->{public_ip} 22") == 0;
+        $timeout -= 1;
+        sleep 5;
+    }
+    die('Cannot reach port 22 on IP ' . $instance->{public_ip}) if ($timeout == 0);
+    return $instance;
+}
+
+=head2 terraform_apply
+
+Calls terraform tool and applies the corresponding configuration .tf file
+
+=cut
+sub terraform_apply {
+    my ($self, %args) = @_;
+    my @instances;
+
+    $args{count}                //= '1';
+    $args{instance_type}        //= get_var('PUBLIC_CLOUD_INSTANCE_TYPE');
```

Comment:
maybe better to have it as local variable ?
```suggestion
   my  $instance_type        //= get_var('PUBLIC_CLOUD_INSTANCE_TYPE');
```

Reply:
> but I do not see much benefit from removing it
since it some new code which just about to be added correct question would be what benefit to add it ? Can you image scenario where job has one `PUBLIC_CLOUD_INSTANCE_TYPE` and we ignoring it and use different one ? Keep in mind that currently `PUBLIC_CLOUD_INSTANCE_TYPE` is defined in Machine level

Reply:
Only where you have some kind of parallel running jobs. Where the SUT is changing it's machine type and the reference not.
But I guess then we need to introduce a new tf file anyway.

Reply:
Thinking twice, @asmorodskyi is right. It's also more easy for now, if we just force the instance type configured by PUBLIC_CLOUD_INSTANCE_TYPE instead of introduce a possible, but not needed complexity.

Reply:
ok, done. We can always make more flexible later on if we need. Thanks for your comments and suggestions.